src/aurelia-binding.d.ts

```diff
@@ -509,6 +512,28 @@ export declare interface InternalCollectionObserver {
    * @param callable A callable object.
    */
   unsubscribe(context: any, callable: Callable): void;
+  /**
+   * Adds a change record to the collection observer.
+   * @param changeRecord 
+   */
+  addChangeRecord(changeRecord: ICollectionObserverSplice): void;
```

Comment:
This isn't necessarily meant to be public API. It requires rather intricate knowledge of the internal data structures and algorithms to be able to use this without messing things up.

Reply:
I completely understand, I can remove this addition.